test/dnnl/TestApiOps.cpp

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cmath>
+#include <gtest/gtest.h>
+
+#include <llvm/ADT/APFloat.h>
+
+#include "DnnlTestUtils.h"
+#include "graph/backend/elyzor/include/dnnl_graph_compiler.h"
+
+static void exec(const char *fileName, dnnl_graph_compiler_tensor *inputs,
+                 dnnl_graph_compiler_tensor *outputs) {
+  auto json = readStrResource(fileName);
+  const struct dnnl_graph_compiler *gc;
+  const struct dnnl_graph_compiler_executable *exe;
+  ASSERT_EQ(dnnl_graph_compiler_create(nullptr, &gc), dnnl_success);
+  ASSERT_EQ(dnnl_graph_compiler_compile(gc, json.c_str(), &exe), dnnl_success);
+  ASSERT_EQ(dnnl_graph_compiler_execute(exe, inputs, outputs), dnnl_success);
+  dnnl_graph_compiler_destroy_executable(exe);
+  dnnl_graph_compiler_destroy(gc);
+}
+
+TEST(TestApiOps, div) {
+  dnnl_graph_compiler_tensor inputs[2];
+  dnnl_graph_compiler_tensor outputs[1];
+  int64_t dims[3] = {64, 32, 32};
+  std::float32_t arg1[64][32][32];
+  std::float32_t arg2[64][32][32];
+  std::float32_t arg3[64][32][32];
+  inputs[0] = {.id = 0, .ndims = 3, .dims = dims, .data = arg1};
+  inputs[1] = {.id = 1, .ndims = 3, .dims = dims, .data = arg2};
+  outputs[0] = {.id = 2, .ndims = 3, .dims = dims, .data = arg3};
+  for (auto i = 0; i < 64; i++) {
+    for (auto j = 0; j < 32; j++) {
+      for (auto k = 0; k < 32; k++) {
+        arg1[i][j][k] = static_cast<std::float32_t>(i * 32 * 32 + j * 32 + k);
+        arg2[i][j][k] = arg1[i][j][k] + 1;
+      }
+    }
+  }
+
+  exec("div.json", inputs, outputs);
+
+  for (auto i = 0; i < 64; i++) {
+    for (auto j = 0; j < 32; j++) {
+      for (auto k = 0; k < 32; k++) {
+        ASSERT_EQ(arg1[i][j][k] / arg2[i][j][k], arg3[i][j][k]);
```

Comment:
shouldn't we use [`ASSERT_NEAR`](http://google.github.io/googletest/reference/assertions.html#EXPECT_NEAR) or something similar everywhere since we're comparing floats?